TestDispatchMetricsShouldDistributeMetrics is flaky

```
--- FAIL: TestDispatchMetricsShouldDistributeMetrics (0.01s)
    handler_backend_test.go:148: 
        	Error Trace:	handler_backend_test.go:148
        	Error:      	Not equal: 
        	            	expected: 304
        	            	actual  : 303
        	Test:       	TestDispatchMetricsShouldDistributeMetrics
    handler_backend_test.go:153: aggregator 0 was invoked 63 time(s)
    handler_backend_test.go:153: aggregator 1 was invoked 62 time(s)
    handler_backend_test.go:153: aggregator 2 was invoked 59 time(s)
    handler_backend_test.go:153: aggregator 3 was invoked 56 time(s)
    handler_backend_test.go:153: aggregator 4 was invoked 63 time(s)
```

Seems to fail 1-2% of the time.  Race related:
```
	wg.Wait()       // Wait for all metrics to be dispatched
	cancelFunc()    // After all metrics have been dispatched, we signal dispatcher to shut down
	wgFinish.Wait() // Wait for dispatcher to shutdown
```

We dispatch all the metrics then cancel the backend as soon as they're in the pipeline.  The backend handler may not have drained the pipeline before we cancel it.

Oddly `TestDispatchMetricMapShouldDistributeMetrics` doesn't seem to have the same failure.  It does a `DispatchMetricMap` on the same goroutine but there's still channels to process it.  It's not immediately obvious what the difference is (it might just be much rarer)

--- a/pkg/statsd/handler_backend_test.go
+++ b/pkg/statsd/handler_backend_test.go
@@ -120,7 +120,8 @@ func TestDispatchMetricsShouldDistributeMetrics(t *testing.T) {
 	r := rand.New(rand.NewSource(time.Now().UnixNano()))
 	n := r.Intn(5) + 1
 	factory := newTestFactory()
-	h := NewBackendHandler(nil, 0, n, 10, factory)
+	// use a sync channel (perWorkerBufferSize = 0) to force the workers to process events before the context is cancelled
+	h := NewBackendHandler(nil, 0, n, 0, factory)
 	ctx, cancelFunc := context.WithCancel(context.Background())
 	defer cancelFunc()
 	var wgFinish wait.Group
@@ -160,7 +161,7 @@ func TestDispatchMetricMapShouldDistributeMetrics(t *testing.T) {
 	r := rand.New(rand.NewSource(time.Now().UnixNano()))
 	numAggregators := r.Intn(5) + 1
 	factory := newTestFactory()
-	// use a sync channel to force the workers to process events before the context is cancelled
+	// use a sync channel (perWorkerBufferSize = 0) to force the workers to process events before the context is cancelled
 	h := NewBackendHandler(nil, 0, numAggregators, 0, factory)
 	ctx, cancelFunc := context.WithCancel(context.Background())
 	defer cancelFunc()

